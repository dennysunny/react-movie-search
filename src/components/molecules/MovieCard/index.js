/* eslint-disable camelcase */
import React from 'react';

import { propTypes, contextTypes } from './propTypes';
import NoImage from '../../../images/noimage.jpg';
import NoBdImage from '../../../images/nobdimage.jpg';
import WatchlistBookmark from '../../atoms/WatchlistBookmark';

const MovieCard = ({ img_base_path, movie, media }, context) => {
  const goToMovie = movieId => (
    context.router.history.push(`/${media}/${movieId}`)
  );

  const {
    id, title, overview, poster_path,
    backdrop_path, vote_average, release_date,
    name, first_air_date,
  } = movie;

  const mtitle = title || name;
  const date = release_date || first_air_date;
  const imagePath = poster_path !== null
    ? [img_base_path + poster_path, img_base_path + backdrop_path]
    : [NoImage, NoBdImage];

  return (
    <li className="movies-list__element">
      <picture
        className="movies-poster"
        onClick={() => goToMovie(id)}
        onKeyDown={() => goToMovie(id)}
      >
        <source
          media="(min-width: 941px)"
          srcSet={imagePath[0]}
        />
        <source
          media="(max-width: 940px)"
          srcSet={imagePath[1]}
        />
        <img
          src={imagePath[0]}
          className="movies-poster"
          alt={mtitle}
        />
      </picture>

      <div className="clearfix">
        <div className="movies-title">
          {mtitle}
        </div>
        <div className="movies-rating">{vote_average}</div>
      </div>
      <div className="clearfix">
        <div className="movies-year">
          {date.slice(0, 4)}
        </div>
      </div>
      <div
        className={[
          'movies-overview ',
          overview.length > 200 ? 'long' : '',
        ].join(' ')}
      >
        {overview.slice(0, overview.indexOf(' ', 200))}
      </div>
      <WatchlistBookmark movie={movie} />
    </li>
  );
};

MovieCard.contextTypes = contextTypes;
MovieCard.propTypes = propTypes;

export default MovieCard;
