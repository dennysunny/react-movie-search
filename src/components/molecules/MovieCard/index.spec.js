import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from '.';

const mockProps = {
  title: 'title',
  release_date: '01.01.2010',
  poster_path: 'path/to/image',
  link: 'url',
  linkCopy: 'lnkCopy',
};

describe('Molecules/MovieCard', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<MovieCard {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
