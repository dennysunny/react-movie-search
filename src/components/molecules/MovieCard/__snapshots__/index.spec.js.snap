// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/MovieCard should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieCard
    img_base_path="path/to/image"
    media="movie"
    movie={
      Object {
        "backdrop_path": "path/to/image",
        "genres": Array [
          Object {
            "genre_name": "genre",
            "id": 1,
          },
        ],
        "id": 1,
        "name": "name",
        "overview": "overview",
        "poster_path": "path/to/image",
        "release_date": "01.01.2010",
        "vote_average": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/movie/1"
        >
          <picture
            className="movies-poster"
          >
            <source
              media="(min-width: 941px)"
              srcSet="path/to/imagepath/to/image"
            />
            <source
              media="(max-width: 940px)"
              srcSet="path/to/imagepath/to/image"
            />
            <img
              alt="name"
              className="movies-poster"
              src="path/to/imagepath/to/image"
            />
          </picture>
        </Link>,
        <div
          className="clearfix"
        >
          <div
            className="movies-title"
          >
            name
          </div>
          <div
            className="movies-rating"
          >
            1
          </div>
        </div>,
        <div
          className="clearfix"
        >
          <div
            className="movies-year"
          >
            01.0
          </div>
        </div>,
        <div
          className="movies-overview  "
        >
          overvie
        </div>,
        <WatchlistBookmark
          movie={
            Object {
              "backdrop_path": "path/to/image",
              "genres": Array [
                Object {
                  "genre_name": "genre",
                  "id": 1,
                },
              ],
              "id": 1,
              "name": "name",
              "overview": "overview",
              "poster_path": "path/to/image",
              "release_date": "01.01.2010",
              "vote_average": 1,
            }
          }
        />,
      ],
      "className": "movies-list__element",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <picture
            className="movies-poster"
          >
            <source
              media="(min-width: 941px)"
              srcSet="path/to/imagepath/to/image"
            />
            <source
              media="(max-width: 940px)"
              srcSet="path/to/imagepath/to/image"
            />
            <img
              alt="name"
              className="movies-poster"
              src="path/to/imagepath/to/image"
            />
          </picture>,
          "replace": false,
          "to": "/movie/1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <source
                media="(min-width: 941px)"
                srcSet="path/to/imagepath/to/image"
              />,
              <source
                media="(max-width: 940px)"
                srcSet="path/to/imagepath/to/image"
              />,
              <img
                alt="name"
                className="movies-poster"
                src="path/to/imagepath/to/image"
              />,
            ],
            "className": "movies-poster",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "media": "(min-width: 941px)",
                "srcSet": "path/to/imagepath/to/image",
              },
              "ref": null,
              "rendered": null,
              "type": "source",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "media": "(max-width: 940px)",
                "srcSet": "path/to/imagepath/to/image",
              },
              "ref": null,
              "rendered": null,
              "type": "source",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "name",
                "className": "movies-poster",
                "src": "path/to/imagepath/to/image",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "picture",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="movies-title"
            >
              name
            </div>,
            <div
              className="movies-rating"
            >
              1
            </div>,
          ],
          "className": "clearfix",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "name",
              "className": "movies-title",
            },
            "ref": null,
            "rendered": "name",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 1,
              "className": "movies-rating",
            },
            "ref": null,
            "rendered": 1,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="movies-year"
          >
            01.0
          </div>,
          "className": "clearfix",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "01.0",
            "className": "movies-year",
          },
          "ref": null,
          "rendered": "01.0",
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "overvie",
          "className": "movies-overview  ",
        },
        "ref": null,
        "rendered": "overvie",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "movie": Object {
            "backdrop_path": "path/to/image",
            "genres": Array [
              Object {
                "genre_name": "genre",
                "id": 1,
              },
            ],
            "id": 1,
            "name": "name",
            "overview": "overview",
            "poster_path": "path/to/image",
            "release_date": "01.01.2010",
            "vote_average": 1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/movie/1"
          >
            <picture
              className="movies-poster"
            >
              <source
                media="(min-width: 941px)"
                srcSet="path/to/imagepath/to/image"
              />
              <source
                media="(max-width: 940px)"
                srcSet="path/to/imagepath/to/image"
              />
              <img
                alt="name"
                className="movies-poster"
                src="path/to/imagepath/to/image"
              />
            </picture>
          </Link>,
          <div
            className="clearfix"
          >
            <div
              className="movies-title"
            >
              name
            </div>
            <div
              className="movies-rating"
            >
              1
            </div>
          </div>,
          <div
            className="clearfix"
          >
            <div
              className="movies-year"
            >
              01.0
            </div>
          </div>,
          <div
            className="movies-overview  "
          >
            overvie
          </div>,
          <WatchlistBookmark
            movie={
              Object {
                "backdrop_path": "path/to/image",
                "genres": Array [
                  Object {
                    "genre_name": "genre",
                    "id": 1,
                  },
                ],
                "id": 1,
                "name": "name",
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "vote_average": 1,
              }
            }
          />,
        ],
        "className": "movies-list__element",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <picture
              className="movies-poster"
            >
              <source
                media="(min-width: 941px)"
                srcSet="path/to/imagepath/to/image"
              />
              <source
                media="(max-width: 940px)"
                srcSet="path/to/imagepath/to/image"
              />
              <img
                alt="name"
                className="movies-poster"
                src="path/to/imagepath/to/image"
              />
            </picture>,
            "replace": false,
            "to": "/movie/1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <source
                  media="(min-width: 941px)"
                  srcSet="path/to/imagepath/to/image"
                />,
                <source
                  media="(max-width: 940px)"
                  srcSet="path/to/imagepath/to/image"
                />,
                <img
                  alt="name"
                  className="movies-poster"
                  src="path/to/imagepath/to/image"
                />,
              ],
              "className": "movies-poster",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "media": "(min-width: 941px)",
                  "srcSet": "path/to/imagepath/to/image",
                },
                "ref": null,
                "rendered": null,
                "type": "source",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "media": "(max-width: 940px)",
                  "srcSet": "path/to/imagepath/to/image",
                },
                "ref": null,
                "rendered": null,
                "type": "source",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "name",
                  "className": "movies-poster",
                  "src": "path/to/imagepath/to/image",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "picture",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="movies-title"
              >
                name
              </div>,
              <div
                className="movies-rating"
              >
                1
              </div>,
            ],
            "className": "clearfix",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "name",
                "className": "movies-title",
              },
              "ref": null,
              "rendered": "name",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
                "className": "movies-rating",
              },
              "ref": null,
              "rendered": 1,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="movies-year"
            >
              01.0
            </div>,
            "className": "clearfix",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "01.0",
              "className": "movies-year",
            },
            "ref": null,
            "rendered": "01.0",
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "overvie",
            "className": "movies-overview  ",
          },
          "ref": null,
          "rendered": "overvie",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "movie": Object {
              "backdrop_path": "path/to/image",
              "genres": Array [
                Object {
                  "genre_name": "genre",
                  "id": 1,
                },
              ],
              "id": 1,
              "name": "name",
              "overview": "overview",
              "poster_path": "path/to/image",
              "release_date": "01.01.2010",
              "vote_average": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
